codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -386,13 +383,57 @@ private void generateOperationHttpBindingSerializer(
     }
 
     private Symbol getRestEncoderSymbol() {
-        return SymbolUtils.createPointableSymbolBuilder("Encoder", GoDependency.AWS_REST_PROTOCOL).build();
+        return SymbolUtils.createPointableSymbolBuilder("Encoder", GoDependency.SMITHY_HTTP_BINDING).build();
     }
 
-    private String generateHttpBindingSetter(
+    private void generateHttpBindingTimestampSerializer(
             Model model,
+            GoWriter writer,
             MemberShape memberShape,
-            String operand
+            HttpBinding.Location location,
+            String operand,
+            BiConsumer<GoWriter, String> locationEncoder
+    ) {
+        writer.addUseImports(GoDependency.SMITHY_TIME);
+
+        TimestampFormatTrait.Format format = memberShape.getMemberTrait(model, TimestampFormatTrait.class)
```

Comment:
You can just call HttpBindingIndex.determineTimestampFormat which is pretty much exactly this

Reply:
I wish there was a better way to discover some of these more obscure functions before I ran off and coded it :D.